python/cuml/pytest_benchmarks/test_ensemble.py

```diff
@@ -0,0 +1,226 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Benchmark integrations described in blog:
+   https://medium.com/rapids-ai/100x-faster-machine-learning-model-ensembling-with-rapids-cuml-and-scikit-learn-meta-estimators-d869788ee6b1
+"""
+
+from cuml.test.utils import unit_param, stress_param
+from cuml.benchmark.algorithms import AlgorithmPair
+from cuml.common.import_utils import has_pytest_benchmark
+
+import pytest
+
+from sklearn.datasets import make_regression, make_classification
+from sklearn.ensemble import AdaBoostRegressor, VotingClassifier
+from sklearn.ensemble import StackingClassifier, BaggingRegressor
+from sklearn.linear_model import LogisticRegression
+from sklearn.neighbors import KNeighborsRegressor
+from sklearn.svm import SVR, SVC
+
+import functools
+
+import cuml
+
+#
+# Testing utilities
+#
+
+
+@functools.lru_cache(maxsize=8)
+def gendata(datafunc, n_samples, n_features, **data_kwargs):
+    return datafunc(n_samples=n_samples, n_features=n_features, **data_kwargs)
+
+
+def _ensemble_benchmark_algo(gpubenchmark,
+                             algorithm_pair,
+                             dataset_func=make_regression,
+                             n_samples=10000,
+                             n_features=100,
+                             data_kwargs={},
+                             algo_type='gpu'):
+    """Execute the benchmark for cpu and gpu, return the Xspeedup as cpu/gpu"""
+
+    data = gendata(dataset_func, n_samples, n_features, **data_kwargs)
+
+    if algo_type == 'cpu':
+        gpubenchmark(algorithm_pair.run_cpu, (data, None))
+    else:
+        gpubenchmark(algorithm_pair.run_cuml, (data, None))
+
+
+def fit_and_score(clf, data, *argc, **kwargs):
+    clf.fit(data[0], data[1])
+    return clf.score(data[-2], data[-1])
+
+
+def fit_and_predict(clf, data):
+    clf.fit(data[0], data[1])
+    return clf.predict(data[0])
+
+#
+# Voting Classifier
+#
+
+
+votingclassifier = AlgorithmPair(
+    cpu_class=VotingClassifier,
+    cuml_class=VotingClassifier,
+    cuml_args={'estimators':
+               [('lr', cuml.linear_model.LogisticRegression(C=1)),
+                ('svc', cuml.svm.SVC())]},
+    cpu_args={'estimators':
+              [('lr', LogisticRegression(C=1, n_jobs=-1)),
+               ('svc', SVC())],
+              'n_jobs': -1},
+    shared_args={'voting': 'hard',
+                 'weights': [0.2, 0.8]},
+    name="VotingClassifier",
+    accepts_labels=False,
+    cpu_data_prep_hook=None,
+    cuml_data_prep_hook=None,
+    accuracy_function=None,
+    bench_func=fit_and_score,
+    setup_cpu_func=None,
+    setup_cuml_func=None)
+
+
+@pytest.mark.integration
+@pytest.mark.skipif(not has_pytest_benchmark(),
+                    reason='pytest-benchmark missing')
+@pytest.mark.parametrize('n_samples', [unit_param(500), stress_param(50000)])
```

Comment:
500 rows seems like it may be too small to be stable. Usually we'd do 10s of thousands of rows as a baseline

Reply:
Agreed the unit_param is more to verify that the methods are working. The stress params were copied from the blog.